### ISSUE TITLE
Understand Office Integration Installer licenses

### ISSUE THREAD

**Issue by cbeach47** [equella/Equella#98]
The code in https://github.com/equella/equella-office-integration is original, except for the office.dll and NetServ.Net.Json.dll under https://github.com/equella/equella-office-integration/tree/master/EquellaOfficeScrapbookIntegration/Lib .  We need to confirm that these two jars are published / licensed appropriately.

To test:
1. Log into an Equella institution.
1. Navigate to 'My Resources' > 'Scrapbook' > 'Upload Files'
1. In the upper right hand corner, click on 'Downloads' to expand the section.
1. Click on 'EQUELLA Office Scrapbook Integration (Windows only)'.

If the dependency is removed, step 4 will result in a 404 file not found.  This is only a download however, so no core functionality is dependent on this dep.

**Comment by cbeach47** [comment]
For NetServ.Net.Json.dll , it looks like it's from https://github.com/andykernahan/netserv-net-json/ .  The license used to be LGPL v2 and then was changed to Apache in 2013.  I haven't found a connection that says the NetServ.Net.Json.dll was built from that source however.

**Comment by bjagg** [comment]
Focusing on NetServ.Net.Json.dll:
@cbeach47 Do you have a reference to where the switch to Apache licensing is documented?

This is not available via Maven; however, if it is Apache we can fork our own version.

**Comment by bjagg** [comment]
Where would someone get the office.dll?

**Comment by cbeach47** [comment]
@bjagg - the NetServ.Net.Json.dll license switch is documented in [this commit](https://github.com/andykernahan/netserv-net-json/commit/3cef87056bbaf03d96100f34c26cb4fbe966e7e3).

I have an internal note out to the team on where the office.dll was originally pulled from.

**Comment by abidingotter** [comment]
I think we could probably use Newtonsoft JSON (MIT license) with minimal code changes if NetServ.Net.Json is problematic. 
As for the office.dll, I'm not sure where that came from initially.  I suspect it may be just an interop DLL to call the real Office DLLs on the user's computer.

**Comment by ddelblanco** [comment]
About the NetServ.Net.Json.dll, based in the size of the file, some information that I've been able to extract from the dll, and all the references to that file in the web that I've found, the original code is the one in the 
https://github.com/andykernahan/netserv-net-json/  that now has the Apache license.  Maybe we have an older version of the file with the previous license... maybe we need to generate a new one with the last code, but I'm not expert in windows development and not sure if that is possible and how.

About the office.dll

Fom all the office.dll versions in the world, (a lot) based in the bytes, this is our guy:
http://systemexplorer.net/file-database/file/office-dll/4453080

That means, as expectd: 
Legal Copyright: 2006 Microsoft Corporation. All rights reserved.
(it can be seen here... and  http://www.processdatabase.org/office-dll-12-0-6425-1000.html

And it is technically called: Core Primary Interop Assembly. So seems and interop, as Aaron said. 

It is a 2007 versions, and now there are 2010 and 2013... and that can be a problem nowadays... not sure if newer office.dll versions will be compatible. 

IMO we should write the instructions to generate the msi based in the code in the repo (removing the office file) + the instructions to find the office.dll file.

I still don't know how to download that from an official Microsoft repo but I've found a place that can be used to download it, So maybe, we can document that this file is needed and people can download and add to the source code themselves....
http://www.fixdll.info/download/58180_office.dll

This forum talks about it: 
https://social.msdn.microsoft.com/Forums/vstudio/en-US/101c03cd-5e3b-44db-b147-189d48a113a9/can-i-redistribute-microsoftofficeinteropoutlookdll-and-officedll-?forum=vsto

Basically someone says that:
"as long a s you have a valid Visual Studio License or a MSDN subscription and you accept the EULA (License Aggreement)you can redistribute and Install the Primary Interop Assemblies with you application"  So we should not provide the msi, each institution should build their own.

**Comment by bjagg** [comment]
Thanks @ddelblanco -- appreciate the research on this!

So for the open source project, we would document as above.

**Comment by ddelblanco** [comment]
I did a PR where we change the link to download the MSI file to a link to the documentation. We should now:
1. Remove the office.dll from the office integration repo (I don't have permissions)
2. Write the documentation about how to add the office.dll to the right place in the code and how to build the msi file.  Is there anyone that has worked before with this and knows how to build it?

**Comment by cbeach47** [comment]
Office.dll was previously removed.  Updated https://github.com/equella/equella-office-integration/blob/master/README.md
